cfme/common/__init__.py

```diff
@@ -365,6 +365,41 @@ def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.policy.item_select('Manage Policies')
 
 
+class ComparableMixin:
+    """
+    Mixin for comparing entities. Should be added to Collection class.
+    Args:
+            provider: An InfraProvider instance
+            entities_list: A list of entities to compare
+    Usage:
+        If added to collection's class,
+
+        entity_coll = collection.all()[:2]
+        compare_view = collection.compare_entities_col(provider, entities_list=entity_coll)
+
+        will navigate to the correct view, select the items in entity_col (first two items returned
+        from all()), click on the compare dropdown item and verify the compare view is
+        displayed. It then returns the compare view.
+    """
+    DROPDOWN_TEXT = 'Compare Selected items'
+    NAV_STRING = 'All'
+
+    def compare_entities_col(self, provider, entities_list=None):
+        from cfme.utils.appliance.implementations.ui import navigate_to
+        entity_view = navigate_to(self, self.NAV_STRING)
+        for item in entities_list:
+            v_entity = entity_view.entities.get_entity(name=item.name)
+            v_entity.ensure_checked()
+        entity_view.toolbar.configuration.item_select(
+            self.DROPDOWN_TEXT, handle_alert=True)
+        if self.parent is provider:
+            compare_entity_view = provider.create_view(self.COMPARE_VIEW)
+        else:
+            compare_entity_view = provider.create_view(self.COMPARE_APP_VIEW)
+        assert compare_entity_view.is_displayed
```

Comment:
I was thinking it would be safer to assert in the mixin so the user wouldn't forget to do it and assume the compare went fine and also I was trying (overly so) to make my test small (one line). But I'll yield to your comment and assume this is the proper pattern used in the framework and we should follow moving forward. I only found 2 instances where assert view_obj.is_displayed was used outside of a test. I'll update as suggested.